### PR TITLE
Be more strict when comparing size in the chunking plugin

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## How Has This Been Tested?
- [x] unit tests
- [ ] @mrow4a can you test this with your instance?

## Screenshots (if appropriate):
This shall not happen
selection_002

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by mrow4a** [comment]
Hmm, still a problem. I use Scality VM, and it is 32bit... I just moved my volumes to 64bit and will see if that fixes the issue.

**Comment by PVince81** [comment]
Fixes https://github.com/owncloud/core/issues/29447

but needs work to make it work on 32 bit systems...

**Comment by PVince81** [comment]
@sharidas did you also have a 32-bit setup somewhere for testing ?

**Comment by sharidas** [comment]
Yes I have a VM now with 32 bit.

**Comment by sharidas** [comment]
Here are the screen shot taken after uploading a ~1 GB ( 977 MB  ) file using desktop client:
filemanager

upload_success_activity_tab

not_synced_no_error

Let me know if this testing can be considered as Ok for 32 bit system. I have linux mint 32 bit xfce running on VMWare Player.

File: apps/dav/lib/Upload/ChunkingPlugin.php [hunk 1]
```diff
@@ -98,7 +101,7 @@ private function verifySize() {
 			return;
 		}
 		$actualSize = $this->sourceNode->getSize();
-		if ((int)$expectedSize !== $actualSize) {
+		if ((int)$expectedSize !== (int)$actualSize) {
```
Comment (sharidas) [inline apps/dav/lib/Upload/ChunkingPlugin.php]: Changing this to `if ($expectedSize != $actualSize) ` ( type juggling ) does work with 32 bit and 64 bit.

**Pull request closed without merging**